data/data/aws/vpc/master-elb.tf

```diff
@@ -46,7 +46,7 @@ resource "aws_lb_target_group" "api_internal" {
   name     = "${var.cluster_id}-aint"
   protocol = "TCP"
   port     = 6443
-  vpc_id   = local.vpc_id
+  vpc_id   = data.aws_vpc.cluster_vpc.id
```

Comment:
using the data in all the clients, ties the clients to how vpc information is fetched.
using resource -> data source makes CREATE, READ, READ api calls which isn't bad but tying our clients to data source to get the vpc id means we can't change those transparently to all the clients.

So i do prefer the local.vpc_id approach.

Reply:
> using resource -> data source makes CREATE, READ, READ api calls...

We're already doing that, aren't we?  And we're already consuming it [here][1] and in other places in master.

> which isn't bad but tying our clients to data source to get the vpc id means we can't change those transparently to all the clients.

We can change this transparently by adjusting what we feed into the `data` block (like [this][2]).  The consumers just look at `data.aws_vpc.cluster_vpc` and don't have to care about whether this is a new or preexisting VPC.

So I don't see any functional difference between `local.vpc_id` and `data.aws_vpc.cluster_vpc.id` besides the latter being slightly slower due to the latency of the fetch we need to run anyway.  And I think adding that latency for consumers that need only the VPC ID is a small price to pay for the benefit of reducing the number of moving parts.  Am I missing something?  Or do you just think that it's worth having local variables to save the latency when the VPC ID is the only thing we care about?

[1]: https://github.com/openshift/installer/blob/1ce819337bbdb96900942101b50205c7f69888e1/data/data/aws/vpc/sg-master.tf#L2
[2]: https://github.com/openshift/installer/blob/93e4c7179a70442208d9e401c7e1df7a2f15cfbd/data/data/aws/vpc/common.tf#L8

Reply:
>  And I think adding that latency for consumers that need only the VPC ID is a small price to pay for the benefit of reducing the number of moving parts. Am I missing something?

it's not super important and we can reconsider if it does later on.